staging/src/k8s.io/pod-security-admission/admission/api/validation/validation.go

```diff
@@ -18,11 +18,75 @@ package validation
 
 import (
 	"k8s.io/apimachinery/pkg/util/validation/field"
-	"k8s.io/pod-security-admission/admission/api"
+	admissionapi "k8s.io/pod-security-admission/admission/api"
+	"k8s.io/pod-security-admission/api"
 )
 
-func ValidatePodSecurityConfiguration(configuration *api.PodSecurityConfiguration) field.ErrorList {
+const (
+	detailValidLevel   = "must be one of 'privileged', 'baseline', or 'restricted'"
+	detailValidVersion = "must be one of 'latest' or 'v<MAJOR>.<MINOR>'"
+)
+
+// ValidatePodSecurityConfiguration validates a given PodSecurityConfiguration.
+func ValidatePodSecurityConfiguration(configuration *admissionapi.PodSecurityConfiguration) field.ErrorList {
 	allErrs := field.ErrorList{}
-	// TODO: validate default levels and versions
+
+	for i, l := range []string{
+		configuration.Defaults.Enforce,
+		configuration.Defaults.Audit,
+		configuration.Defaults.Warn,
+	} {
+		_, err := api.ParseLevel(l)
+		if err != nil {
+			path := labelPath(labelForLevelAtIndex((i)))
```

Comment:
This is the field path for setting the policy on a namespace, but for the configuration this should use the actual fieldpath for the configuration (e.g. `field.NewPath("configuration", "defaults", "enforce")`)

Reply:
yes, field path from the root of the config (e.g. NewPath("defaults","enforce"))

Reply:
it's usually simpler to have a method like `validatePodSecurityDefaults(fieldPath *field.Path, defaults PodSecurityDefaults) field.ErrorList` and call it three times like this:

```go
allErrs = append(allErrs, validatePodSecurityDefaults(field.NewPath("defaults","enforce"), configuration.Defaults.Enforce)...)
allErrs = append(allErrs, validatePodSecurityDefaults(field.NewPath("defaults","warn"), configuration.Defaults.Warn)...)
allErrs = append(allErrs, validatePodSecurityDefaults(field.NewPath("defaults","audit"), configuration.Defaults.Audit)...)
```

Reply:
huh... or at least it would have been if we had structured the config that way :)

Reply:
try something like this:
```
allErrs = append(allErrs, validateLevel(field.NewPath("defaults","enforce"), configuration.Defaults.Enforce)...)
allErrs = append(allErrs, validateVersion(field.NewPath("defaults","enforce-version"), configuration.Defaults.EnforceVersion)...)

allErrs = append(allErrs, validateLevel(field.NewPath("defaults","warn"), configuration.Defaults.Warn)...)
allErrs = append(allErrs, validateVersion(field.NewPath("defaults","warn-version"), configuration.Defaults.WarnVersion)...)

allErrs = append(allErrs, validateLevel(field.NewPath("defaults","audit"), configuration.Defaults.Audit)...)
allErrs = append(allErrs, validateVersion(field.NewPath("defaults","audit-version"), configuration.Defaults.AuditVersion)...)
```

Reply:
that lets you skip the labelFor_AtIndex indirection